open_humans/templatetags/utilities.py

```diff
@@ -298,3 +324,37 @@ def add_string(a, b):
     Like `add` but coerces to strings instead of integers.
     """
     return str(a) + str(b)
+
+
+@register.tag
+def render_if_visible(parser, token):
+    """
+    Is a member publicly sharing data but wishes that membership to not be public?
+    """
+    try:
+        tag, user_t, project_t = token.split_contents()
+    except ValueError:
+        raise template.TemplateSyntaxError("is_visible requires exactly two arguments")
+
+    nodelist = parser.parse(('end_render_if_visible',))
+    parser.delete_first_token()
+
+    user = parser.compile_filter(user_t)
+    project = parser.compile_filter(project_t)
+
+    return VisibleNode(nodelist, user, project)
+
+
+class VisibleNode(template.Node):
+    def __init__(self, nodelist, user, project):
+        self.nodelist = nodelist
+        self.user = user
+        self.project = project
+
+    def render(self, context):
+        user = self.user.resolve(context)
+        project = self.project.resolve(context)
+        if project_membership_visible(user, project):
```

Comment:
I think what's actually passed in the code at this point is a `member` despite it being called a `user` here – in which case the code variables should be renamed accordingly. I think it'd also be more clear to call it a `source` or `source_label` to be more clear that it's not a `DataRequestProject` object.